### PR TITLE
Check database exist before create

### PR DESCRIPTION
**Pull request by fogelito**

### CODE REVIEW

File: tests/Database/Adapter/MariaDBTest.php [hunk 1]
```diff
@@ -49,11 +49,6 @@ public static function getDatabase(): Database
         $database = new Database(new MariaDB($pdo), $cache);
         $database->setDefaultDatabase('utopiaTests');
         $database->setNamespace('myapp_'.uniqid());
-
-        if ($database->exists('utopiaTests')) {
```
Comment (fogelito) [inline tests/Database/Adapter/MariaDBTest.php:53]: I remove this,  since it is Done in Base.php Tests

**Comment by abnegate** [comment]
Do we need this change? Shouldn't it be okay as is with `CREATE DATABASE IF NOT EXISTS`?

**Comment by fogelito** [comment]
We had some lock incidents with the create schema, it is triggering a DDL operation,
https://stackoverflow.com/questions/6169263/mysqls-create-database-if-not-exists-misbehaves-when-database-has-live-connec
@stnguyen90 What do you think about it? you saw this live

**Review by stnguyen90 — APPROVED** [review]

**Comment by abnegate** [comment]
> We had some lock incidents with the create schema, it is triggering a DDL operation, https://stackoverflow.com/questions/6169263/mysqls-create-database-if-not-exists-misbehaves-when-database-has-live-connec @stnguyen90 What do you think about it? you saw this live

@fogelito I don't think this query was the one that held the lock, but the one that was blocked when trying to acquire it

**Comment by stnguyen90** [comment]
> @fogelito I don't think this query was the one that held the lock, but the one that was blocked when trying to acquire it

@abnegate I think this can make the problem worse because once this query starts, a full database lock is created and can block other queries after.

File: tests/Database/Adapter/MariaDBTest.php (hunk 1 above)
Comment (abnegate) [reply tests/Database/Adapter/MariaDBTest.php:53]: Can you link it? I don't see a call to delete in Base.php

Comment (fogelito) [reply tests/Database/Adapter/MariaDBTest.php:53]: https://github.com/utopia-php/database/blob/107a6332c01da5c08bc2b31a158cf08acc0ac792/tests/Database/Base.php#L72

I guess I can revert this to be more consistent for a fresh start, not to depend on this exist method 👍

**Review by eldadfux — APPROVED** [review]

File: src/Database/Adapter/MariaDB.php [hunk 2]
```diff
@@ -27,6 +27,10 @@ public function create(string $name): bool
     {
         $name = $this->filter($name);
 
+        if ($this->exists($name)) {
+            return true;
+        }
+
         $sql = "CREATE DATABASE IF NOT EXISTS `{$name}` /*!40100 DEFAULT CHARACTER SET utf8mb4 */;";
```
Comment (eldadfux) [inline src/Database/Adapter/MariaDB.php:34]:
```suggestion
        $sql = "CREATE DATABASE `{$name}` /*!40100 DEFAULT CHARACTER SET utf8mb4 */;";
```

With the new check we don't really need this.

Comment (eldadfux) [reply src/Database/Adapter/MariaDB.php:34]: Also, do we have the same in other adapters?

**Review by abnegate — APPROVED** [review]

**Pull request merged**